dali/kernels/erase/erase_gpu.h

```diff
@@ -503,24 +501,22 @@ struct EraseGpu {
       auto &sample = sample_desc_cpu[i];
       sample.in = in.data[i];
       sample.out = out.data[i];
-      sample.erase_regions = erased_regions.data[i];
+      sample.erase_regions = make_cspan(erased_regions.tensor_data(i),
+                                        erased_regions.tensor_shape(i).num_elements());
       for (int dim = 0; dim < ndim; dim++) {
         sample.sample_shape[dim] = in.tensor_shape_span(i)[dim];
       }
       sample.sample_stride = GetStrides(sample.sample_shape);
     }
 
-    CUDA_CALL(
-      cudaMemcpyAsync(sample_desc_gpu, sample_desc_cpu, num_samples * sizeof(sample_t),
-                      cudaMemcpyHostToDevice, stream));
-    CUDA_CALL(
-      cudaMemcpyAsync(fill_values_gpu, fill_values_cpu, num_fill_values * sizeof(T),
-                      cudaMemcpyHostToDevice, stream));
+    auto *sample_desc_gpu = ctx.scratchpad->ToGPU(stream, make_span(sample_desc_cpu, num_samples));
+    auto* fill_values_gpu =
+      ctx.scratchpad->ToGPU(stream, make_span(fill_values_cpu, num_fill_values));
```

Comment:
Better option would be to use `ToContiguousGPU` - it will issue just one cudaMemcpy.

It's not in the scope of this task, I guess, however, if we keep some sane limit on number of channels, then the fill value could be copied to a `__constant__` - it should improve the performance, since the fill_value won't have to be read from global memory and will not compete for cache with input.

Reply:
Used ToContiguousGPU